core/drivers/gpio/gpio.c

```diff
@@ -32,14 +32,17 @@ TEE_Result gpio_dt_alloc_pin(struct dt_pargs *pargs, struct gpio **out_gpio)
 
 static char *gpio_get_dt_prop_name(const char *gpio_name)
 {
+	int ret = 0;
 	char *prop_name = NULL;
-	int max_len = strlen(gpio_name) + strlen("-gpios") + 1;
+	size_t max_len = strlen(gpio_name) + strlen("-gpios") + 1;
 
 	prop_name = calloc(1, max_len);
 	if (!prop_name)
 		return NULL;
 
-	snprintf(prop_name, max_len, "%s-gpios", gpio_name);
+	ret = snprintf(prop_name, max_len, "%s-gpios", gpio_name);
+	if (ret < 0 || (size_t)ret >= max_len)
+		return NULL;
```

Comment:
You need to free `prop_name` before returning.

Reply:
Oh right